packages/dataviews/src/dataform-controls/utils/validated-input.tsx

```diff
@@ -54,6 +54,15 @@ export default function ValidatedText< Item >( {
 		[ data, setValue, onChange ]
 	);
 
+	// Convert pattern to string if it's a RegExp
+	let pattern;
```

Comment:
In addition to per-control validation, the `useFormValidity` hook must return the same error in the `validity` prop and `isValid` false. This is demonstrated in the storybook by controlling the "save button" (it's disabled if the form is invalid).